Libraries/Components/TextInput/TextInput.js

```diff
@@ -768,8 +768,8 @@ const TextInput = React.createClass({
     // that we have in JS.
     const nativeProps = {};
 
-    if (this._lastNativeText !== this.props.value && typeof this.props.value === 'string') {
-      nativeProps.text = this.props.value;
+    if (this._lastNativeText !== this.props.value && typeof this.props.value !== 'string') {
```

Comment:
This will break controlled textinput components, where the user changes the value of `props.value` in response to the user inputting some text.